### ISSUE TITLE
Compatibility with PL 2.0 and Pytorch 2.0

### ISSUE THREAD

**Issue by surajpaib** [project-lighter/lighter#51]
This issue is to track changes to make `lighter` compatible with PL2.0 and Pytorch 2.0

**Comment by ibro45** [comment]
Do we still need this?

**Comment by ibro45** [comment]
Looked at https://lightning.ai/docs/pytorch/latest/upgrade/from_1_9.html

---
### Issue 1
<img width="920" alt="image" src="https://user-images.githubusercontent.com/18015788/228571889-830c5d09-b6ad-41d6-91f8-e1a291c4c9a6.png">

https://github.com/project-lighter/lighter/blob/5821d1921cc2da4252c7d0ea8f5e7a8f8f343be9/lighter/utils/cli.py#L26
`.tune()` is part of the `Tuner` only now and removed from the `Trainer.

--- 
### Issue 2
<img width="920" alt="image" src="https://user-images.githubusercontent.com/18015788/228573946-869aafd9-f6ac-4ef9-9751-a2bb88d80429.png">

Previously, I had issues with logging when using gradient accumulation. That's because the step counter inside of the Trainer/LightningModule (can't remember which) would count a step only when an `optimizer.step()` happens, and that doesn't happen until `n` batch gradients are accumulated. That's why I did this in the logger:
https://github.com/project-lighter/lighter/blob/5821d1921cc2da4252c7d0ea8f5e7a8f8f343be9/lighter/callbacks/logger.py#L44
Not sure if that's still necessary because of that. If it's not, maybe adjust it, but also keep an eye on this https://github.com/project-lighter/lighter/blob/5821d1921cc2da4252c7d0ea8f5e7a8f8f343be9/lighter/callbacks/logger.py#L288

---
